src/WebJobs.Script.WebHost/Security/ISecretManager.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.Azure.WebJobs.Host;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost
+{
+    public interface ISecretManager
+    {
+        bool DeleteSecret(string secretName, string functionName = null);
+
+        Dictionary<string, string> GetFunctionSecrets(string functionName);
+
+        Dictionary<string, string> GetMergedFunctionSecrets(string functionName);
```

Comment:
Wonder if you shouldn't just add an optional param "merged" on GetFunctionSecrets so both scenarios can be handled with the single function?